linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1274,18 +1282,16 @@ private[optimizer] abstract class OptimizerCore(
       case _ =>
         def default: TailRec[Tree] = {
           resolveLocalDef(preTransQual) match {
-            case PreTransRecordTree(newQual, origType, cancelFun) =>
-              val recordType = newQual.tpe.asInstanceOf[RecordType]
-              /* FIXME How come this lookup requires only the `simpleName`?
-               * The `recordType` is created at `InlineableClassStructure.recordType`,
-               * where it uses an allocator. Something fishy is going on here.
-               * (And no, this is not dead code.)
-               */
-              val recordField = recordType.findField(field.name.simpleName)
+            case PreTransRecordTree(newQual, structure, cancelFun) =>
+              val recordField = structure.findField(field.name)
               val sel = RecordSelect(newQual, SimpleFieldIdent(recordField.name))(recordField.tpe)
               sel.tpe match {
                 case _: RecordType =>
-                  cont(PreTransRecordTree(sel, RefinedType(expectedType), cancelFun))
+                  /* We're trying to select a reified subrecord. We lost its
+                   * structure so cannot build an accurate `PreTransRecordTree`.
+                   * This did not happen in our test suite at the time of writing.
+                   */
```

Comment:
Just to check my understanding: This would/could happen with nested class inlining (if a field of an `@inline` class is itself of an `@inline` type).

Reply:
Yes. If we have nested class inlining, *and* we reify the outermost as a record (which only happens in the case of branch merging), then we can end up in this situation.